lib/util/Components.js

```diff
@@ -179,6 +181,53 @@ class Components {
     const list = Lists.get(this);
     return Object.keys(list).filter((i) => list[i].confidence >= 2).length;
   }
+
+  /**
+   * Return the node naming the default React import
+   * It can be used to determine the local name of import, even if it's imported
+   * with an unusual name.
+   *
+   * @returns {ASTNode} React default import node
+   */
+  getDefaultReactImport() {
+    const info = ReactImports.get(this);
+    return info.defaultReactImport;
+  }
+
+  /**
+   * Return the nodes of all React named imports
+   *
+   * @returns {Object} The list of React named imports
+   */
+  getNamedReactImports() {
+    const info = ReactImports.get(this);
+    return info.namedReactImports;
+  }
+
+  /**
+   * Add the default React import specifier to the scope
+   *
+   * @param {ASTNode} specifier The AST Node of the default React import
+   * @returns {void}
+   */
+  addDefaultReactImport(specifier) {
+    const info = ReactImports.get(this);
+    info.defaultReactImport = specifier;
```

Comment:
```suggestion
    ReactImports.get(this).defaultReactImport = specifier;
```
altho i'm not a fan of the mutation here. why not `ReactImports.set(this, { ...info, defaultReactImport: specifier })`?

also tho, when would this function be called when there's already a defaultReactImport? Should it throw if it's called when `defaultReactImport` is already set?

Reply:
It could throw, but it seems like it should be another rule's responsibility to simply flag the double-import, not this rules responsibility to throw a fit.

In normal cases the first / only definition is detected.
In abnormal cases the final detected instance wins; a little weird but at least predictable.

Reply:
A rule couldn’t handle that unless this utility tracked every occurrence.

It seems like it’d be better then to keep an array, and rules can assume there’s one, or check, as they like.

Reply:
I went ahead and made `defaultReactImports` an accumulating array the same as `namedReactImports`, and assert the same way using `deepEqual` over the mapped local identifiers of each of the returned arrays.